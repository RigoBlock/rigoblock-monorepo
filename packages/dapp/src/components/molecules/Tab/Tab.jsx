import './Tab.scss'
import PropTypes from 'prop-types'
import React from 'react'
import WrapperWithDivider from '../../molecules/WrapperWithDivider'

const divider = () => <div className="modal-divider" />

const Tab = ({ Header, forms }) => {
  if (forms.length === 1) {
    const Form = forms[0]
    return (
      <div className="form-modal">
        <WrapperWithDivider Divider={divider}>
          <div className="form-header">
            <Header />
          </div>
          <div className="form-content">
            <Form />
          </div>
        </WrapperWithDivider>
      </div>
    )
  }
  // TODO: add logic to support multiple forms
}

Tab.propTypes = {
  forms: PropTypes.array.isRequired,
  Header: PropTypes.func.isRequired
}

export default Tab
