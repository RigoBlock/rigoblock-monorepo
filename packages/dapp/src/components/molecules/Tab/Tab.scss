@import "../../_settings/variables.scss";

.tab {
  max-width: 9.25 * $dist-l;
  background: $secondary-bg-color;

  .tab-header, .tab-content {
    padding: $dist-s;
  }

  .tab-divider {
    background: $dark-grey;
    height: 1px;
  }
}

