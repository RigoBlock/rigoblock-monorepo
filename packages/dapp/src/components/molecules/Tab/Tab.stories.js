import '../../_settings/_base.scss'
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import { storiesOf } from '@storybook/react'
import React from 'react'
import Tab from './'
import Title from '../../atoms/Title'
import mockStore from '../../../fixtures/store'

const store = mockStore()

const title = () => <Title>Tab Title</Title>
const form = () => <div>Placeholder Form</div>

storiesOf('Molecules/Tab', module)
  .addDecorator(story => <Provider store={store}>{story()}</Provider>)
  .addDecorator(story => <MemoryRouter>{story()}</MemoryRouter>)
  .add('default', () => <Tab Header={title} forms={[form]} />)
