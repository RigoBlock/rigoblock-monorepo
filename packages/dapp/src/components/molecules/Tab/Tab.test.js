import React from 'react'
import Tab from './Tab.jsx'
import toJson from 'enzyme-to-json'

const formCreatorFunction = () => (
  <div key="test component">Header Component</div>
)
const Header = () => <div>Header Component</div>
const props = {
  Header,
  forms: [formCreatorFunction]
}

describe('Tab component', () => {
  it('renders correctly', () => {
    expect(toJson(createComponentWithProps(Tab, props))).toMatchSnapshot()
  })
})
