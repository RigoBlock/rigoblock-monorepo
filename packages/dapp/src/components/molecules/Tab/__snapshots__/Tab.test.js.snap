// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab component renders correctly 1`] = `undefined`;
